src/backend/distributed/transaction/transaction_recovery.c

```diff
@@ -404,7 +404,7 @@ PendingWorkerTransactionList(MultiConnection *connection)
 	int32 coordinatorId = GetLocalGroupId();
 
 	appendStringInfo(command, "SELECT gid FROM pg_prepared_xacts "
-							  "WHERE gid LIKE 'citus\\_%d\\_%%'",
+							  "WHERE gid LIKE 'citus\\_%d\\_%%' and database = current_database()",
```

Comment:
thanks, this makes a lot of sense. (and slightly embarrassing that we did not do this before)